crates/aptos-dkg/src/range_proofs/univariate_range_proof.rs

```diff
@@ -367,35 +366,47 @@ where
 
     // Step 6:
     //  1. Compute each f_j(X) in coefficient form via a size-(n+1) FFT on f_j(X)
-    //  2. Comptue f'_j(X) via a differentiation.
+    //  2. Compute f'_j(X) via a differentiation.
     //  3. Evaluate f'_j at all (n+1)th roots of unity via a size-(n+1) FFT.
     //  4. \forall i \in [0,n), compute N_j'(\omega^i) = (\omega^i - \omega^n) f_j'(\omega^i)(2f_j(\omega^i) - 1)
     //  5. for i = n, compute N_j'(\omega^n) = r_j(r_j - 1)
     //  6. \forall i \in [0,n], compute h_j(\omega^i) = N_j'(\omega^i) / ( (n+1)\omega^{i n} )
     #[cfg(feature = "range_proof_timing")]
     let start = Instant::now();
-    let omega_n = pp.batch_dom_n1.get_root_of_unity(pp.n);
-    let n1_inv = Scalar::from(pp.n as u64 + 1).invert().unwrap();
+    let omega_n = pp.eval_dom.element(pp.n); // let omega_n = pp.all_roots_of_unity(pp.n)
```

Comment:
This may unnecessarily compute it? Or does it fetch it? Cause you do have it pre-computed in `roots_of_unity_in_eval_dom`, no?